ideascube/serveradmin/catalog.py

```diff
@@ -611,8 +611,20 @@ def list_remotes(self):
         return sorted(self._remotes.values(), key=attrgetter('id'))
 
     def add_remote(self, id, name, url):
-        if id in self._remotes:
-            raise ValueError('There already is a "{}" remote'.format(id))
+        try:
+            existing_remote = self._remotes[id]
+            if url != existing_remote.url:
+                raise ValueError(
+                         ('There already is a "{0.id}" remote'
+                          'and urls differ ({0.url} and {1}'
+                         ).format(existing_remote, url)
+                      )
+
+            print(('CommandWarning: "{}" remote already exists.\n'
```

Comment:
Not sure about "CommandWarning" prefix. I'd say simply "Warning". Or we should a consistent usage in all messages, no?